relay-profiling/src/lib.rs

```diff
@@ -108,36 +110,86 @@ use crate::android::expand_android_profile;
 use crate::cocoa::expand_cocoa_profile;
 use crate::python::parse_python_profile;
 use crate::rust::parse_rust_profile;
+use crate::sample::expand_sample_profile;
 use crate::typescript::parse_typescript_profile;
 
 pub use crate::error::ProfileError;
 
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "lowercase")]
+enum Platform {
+    Android,
+    Cocoa,
+    Node,
+    Python,
+    Rust,
+    Typescript,
+}
+
 #[derive(Debug, Deserialize)]
 struct MinimalProfile {
-    platform: String,
+    platform: Platform,
+    version: Option<String>,
 }
 
 fn minimal_profile_from_json(data: &[u8]) -> Result<MinimalProfile, ProfileError> {
     serde_json::from_slice(data).map_err(ProfileError::InvalidJson)
 }
 
 pub fn expand_profile(payload: &[u8]) -> Result<Vec<Vec<u8>>, ProfileError> {
-    let minimal_profile: MinimalProfile = minimal_profile_from_json(payload)?;
-    let platform = minimal_profile.platform.as_str();
-    match platform {
-        "android" => expand_android_profile(payload),
-        "cocoa" => expand_cocoa_profile(payload),
-        _ => {
-            let payload = match platform {
-                "python" => parse_python_profile(payload),
-                "rust" => parse_rust_profile(payload),
-                "typescript" => parse_typescript_profile(payload),
-                _ => Err(ProfileError::PlatformNotSupported),
-            };
-            match payload {
-                Ok(payload) => Ok(vec![payload]),
-                Err(payload) => Err(payload),
+    let profile: MinimalProfile = minimal_profile_from_json(payload)?;
+    match profile.version {
+        Some(_) => expand_sample_profile(payload),
+        None => match profile.platform {
+            Platform::Android => expand_android_profile(payload),
+            Platform::Cocoa => expand_cocoa_profile(payload),
+            _ => {
+                let payload = match profile.platform {
+                    Platform::Python => parse_python_profile(payload),
+                    Platform::Rust => parse_rust_profile(payload),
+                    Platform::Typescript => parse_typescript_profile(payload),
+                    _ => Err(ProfileError::PlatformNotSupported),
+                };
+                match payload {
+                    Ok(payload) => Ok(vec![payload]),
+                    Err(err) => Err(err),
+                }
```

Comment:
```suggestion
                Ok(vec![payload?])
```
This might be simpler and there is no need for another `match` here